[Bug] Won't run (part 2)

Will open new issue as previous one closed and no responses - https://github.com/Alex8791-cyber/cognithor/issues/115

Just tried installing 0.92.4 via windows installer - CognithorSetup-0.92.4.exe

Installed fine, but won't run, icon near clock shows status: stopped for backend, restarting it and it goes back to stopped.

Running cognithor.bat gives error as below:

[OK] Python: C:\Users\darre\AppData\Local\Cognithor\python\python.exe
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\darre\AppData\Local\Cognithor\python\Lib\site-packages\cognithor\__main__.py", line 2370, in <module>
    main()
  File "C:\Users\darre\AppData\Local\Cognithor\python\Lib\site-packages\cognithor\__main__.py", line 405, in main
    config = load_config(args.config)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\darre\AppData\Local\Cognithor\python\Lib\site-packages\cognithor\config.py", line 3244, in load_config
    cfg = CognithorConfig(**data)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\darre\AppData\Local\Cognithor\python\Lib\site-packages\pydantic\main.py", line 263, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pydantic_core._pydantic_core.ValidationError: 4 validation errors for CognithorConfig
max_agents
  Extra inputs are not permitted [type=extra_forbidden, input_value=1, input_type=int]
    For further information visit https://errors.pydantic.dev/2.13/v/extra_forbidden
max_concurrent
  Extra inputs are not permitted [type=extra_forbidden, input_value=1, input_type=int]
    For further information visit https://errors.pydantic.dev/2.13/v/extra_forbidden
memory_limit_mb
  Extra inputs are not permitted [type=extra_forbidden, input_value=2048, input_type=int]
    For further information visit https://errors.pydantic.dev/2.13/v/extra_forbidden
rag
  Extra inputs are not permitted [type=extra_forbidden, input_value={'enabled': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.13/v/extra_forbidden

[ERROR] Cognithor exited with an error.
Press any key to continue . . .

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,7 +5,7 @@ All notable changes to Cognithor are documented in this file.
 Format based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 Versioning follows [Semantic Versioning](https://semver.org/).
 
-## [unreleased]
+## [Unreleased]
 
 ### Added
 - **Qwen3.6 model registry + installer** (`cognithor models install <name>`).
@@ -41,6 +41,19 @@ Versioning follows [Semantic Versioning](https://semver.org/).
   format the current Flutter client has never sent. File uploads from
   Flutter were silently dropped. The gate is now `metadata.file_base64`
   presence, covering the real message shape.
+
+### Fixed
+- **`config.yaml` with legacy keys no longer hard-crashes at startup**
+  (closes #131). Upgrading users whose `config.yaml` still contains fields
+  the current `CognithorConfig` no longer recognizes (e.g. `max_agents`,
+  `max_concurrent`, `memory_limit_mb`, `rag`) would hit
+  `pydantic_core.ValidationError: extra_forbidden` inside `load_config`
+  and be completely unable to launch. `load_config` now catches
+  `extra_forbidden` errors from the on-disk YAML read path, strips the
+  offending keys, logs a clear WARN listing them, and re-validates once.
+  `CognithorConfig` keeps `extra="forbid"` on the model itself so
+  programmatic construction in tests still catches typos — only the
+  user-supplied YAML is self-healing.
 
 ### Deferred
 - **Video input** is explicitly deferred. Qwen3.6-27B (VLM) can process

--- a/src/cognithor/config.py
+++ b/src/cognithor/config.py
@@ -19,7 +19,7 @@ from pathlib import Path
 from typing import Any, Literal
 
 import yaml
-from pydantic import BaseModel, ConfigDict, Field, field_validator, model_validator
+from pydantic import BaseModel, ConfigDict, Field, ValidationError, field_validator, model_validator
 
 from cognithor.models import ModelConfig, SandboxConfig
 
@@ -3240,13 +3240,58 @@ def load_config(config_path: Path | None = None) -> CognithorConfig:
             if isinstance(val, str):
                 data["models"][role] = {"name": val}
 
-    # 6. Pydantic validates and fills defaults
-    cfg = CognithorConfig(**data)
+    # 6. Pydantic validates and fills defaults.
+    #    Legacy/unknown YAML keys (e.g. from old Jarvis versions) are stripped
+    #    with a warning instead of crashing the whole launch (see issue #131).
+    #    `extra="forbid"` is kept on the model so programmatic construction in
+    #    tests still catches typos — only user-supplied YAML is self-healing.
+    try:
+        cfg = CognithorConfig(**data)
+    except ValidationError as exc:
+        extra_errors = [e for e in exc.errors() if e.get("type") == "extra_forbidden"]
+        if not extra_errors:
+            raise
+        stripped = _strip_extra_forbidden_keys(data, extra_errors)
+        if stripped:
+            import logging
+
+            logging.getLogger("cognithor.config").warning(
+                "Unbekannte Felder in config.yaml ignoriert (bitte aus der Datei entfernen): %s",
+                ", ".join(stripped),
+            )
+        cfg = CognithorConfig(**data)
 
     # 7. Keyring fallback: fill empty API-key fields from OS Keyring
     _resolve_secrets(cfg)
 
     return cfg
+
+
+def _strip_extra_forbidden_keys(
+    data: dict[str, Any], extra_errors: list[dict[str, Any]]
+) -> list[str]:
+    """Remove keys flagged as ``extra_forbidden`` from ``data`` in-place.
+
+    Navigates the nested ``loc`` path from each Pydantic error and drops the
+    leaf key. Returns the dotted paths of keys actually removed so the caller
+    can report them to the user.
+    """
+    stripped: list[str] = []
+    for err in extra_errors:
+        loc = err.get("loc") or ()
+        if not loc:
+            continue
+        parent: Any = data
+        for key in loc[:-1]:
+            if not isinstance(parent, dict) or key not in parent:
+                parent = None
+                break
+            parent = parent[key]
+        leaf = loc[-1]
+        if isinstance(parent, dict) and leaf in parent:
+            del parent[leaf]
+            stripped.append(".".join(str(k) for k in loc))
+    return stripped
 
 
 def _resolve_secrets(config: CognithorConfig) -> None:

--- a/tests/config/test_legacy_keys.py
+++ b/tests/config/test_legacy_keys.py
@@ -1,0 +1,68 @@
+from __future__ import annotations
+
+import logging
+
+import pytest
+from pydantic import ValidationError
+
+from cognithor.config import CognithorConfig, load_config
+
+
+class TestLegacyTopLevelKeysAreTolerated:
+    """Legacy YAML keys must not crash `load_config` — issue #131.
+
+    A user upgrading from an older Jarvis version can have a `config.yaml`
+    with fields the current `CognithorConfig` no longer recognizes
+    (e.g. `max_agents`, `memory_limit_mb`, `rag`). `extra="forbid"` on
+    the model used to turn these into an unrecoverable launch crash.
+    """
+
+    def test_unknown_top_level_keys_are_stripped_with_warning(self, tmp_path, caplog):
+        cfg_file = tmp_path / "config.yaml"
+        cfg_file.write_text(
+            "max_agents: 1\n"
+            "max_concurrent: 1\n"
+            "memory_limit_mb: 2048\n"
+            "rag:\n"
+            "  enabled: false\n"
+            "language: de\n",
+            encoding="utf-8",
+        )
+
+        with caplog.at_level(logging.WARNING, logger="cognithor.config"):
+            cfg = load_config(cfg_file)
+
+        assert cfg is not None
+        assert cfg.language == "de"
+        warnings = " ".join(r.getMessage() for r in caplog.records)
+        for key in ("max_agents", "max_concurrent", "memory_limit_mb", "rag"):
+            assert key in warnings, (
+                f"Expected key '{key}' in deprecation warning, got: {warnings!r}"
+            )
+
+    def test_valid_config_without_unknown_keys_still_loads_clean(self, tmp_path, caplog):
+        cfg_file = tmp_path / "config.yaml"
+        cfg_file.write_text("language: en\nowner_name: Tester\n", encoding="utf-8")
+
+        with caplog.at_level(logging.WARNING, logger="cognithor.config"):
+            cfg = load_config(cfg_file)
+
+        assert cfg.language == "en"
+        assert cfg.owner_name == "Tester"
+        unknown_warnings = [r for r in caplog.records if "Unbekannte Felder" in r.getMessage()]
+        assert unknown_warnings == []
+
+    def test_programmatic_construction_still_rejects_unknown_fields(self):
+        """The safety net on the model itself is preserved — dev/test code
+        paths that construct `CognithorConfig(**kwargs)` directly must keep
+        catching typos, only the on-disk YAML read path is tolerant."""
+        with pytest.raises(ValidationError):
+            CognithorConfig(definitely_not_a_field=1)
+
+    def test_real_errors_still_surface_unchanged(self, tmp_path):
+        """When the YAML has a *real* error (wrong type on a known field),
+        `load_config` must still raise — we only swallow extra_forbidden."""
+        cfg_file = tmp_path / "config.yaml"
+        cfg_file.write_text("language: 123\n", encoding="utf-8")
+        with pytest.raises(ValidationError):
+            load_config(cfg_file)

